GoogleBiddingAdapter/src/main/java/com/chartboost/mediation/googlebiddingadapter/GoogleBiddingAdapter.kt

```diff
@@ -1123,22 +1039,214 @@ class GoogleBiddingAdapter : PartnerAdapter {
+            if (it.isActive) {
+                it.resume(Result.success(partnerAdWeakRef.get() ?: return))
+            }
+        }
+    }
+
+    override fun onAdClicked() {
+        PartnerLogController.log(DID_CLICK)
+
+        listenerWeakRef.get()?.onPartnerAdClicked(partnerAdWeakRef.get() ?: return)
+            ?: PartnerLogController.log(
+                CUSTOM,
+                "Unable to fire onPartnerAdClicked for Google Bidding adapter.",
+            )
+    }
+
+    override fun onAdDismissedFullScreenContent() {
+        PartnerLogController.log(DID_DISMISS)
+        listenerWeakRef.get()?.onPartnerAdDismissed(partnerAdWeakRef.get() ?: return, null)
+            ?: PartnerLogController.log(
+                CUSTOM,
+                "Unable to fire onPartnerAdDismissed for Google Bidding adapter.",
+            )
+    }
+}
+
+/**
+ * Callback class for rewarded ads.
+ *
+ * @param listenerRef A [WeakReference] to the [PartnerAdListener] to be notified of ad events.
+ * @param partnerAdRef A [WeakReference] to the [PartnerAd] object containing the Google Bidding ad to be shown.
+ * @param continuationRef A [WeakReference] to the [CancellableContinuation] to be resumed once the ad is shown.
+ */
+private class RewardedAdShowCallback(
+    listenerRef: WeakReference<PartnerAdListener?>,
+    partnerAdRef: WeakReference<PartnerAd>,
+    continuationRef: WeakReference<CancellableContinuation<Result<PartnerAd>>>,
+) : FullScreenContentCallback() {
+    private val listenerWeakRef = listenerRef
+    private val partnerAdWeakRef = partnerAdRef
+    private val continuationWeakRef = continuationRef
+
+    override fun onAdImpression() {
+        PartnerLogController.log(DID_TRACK_IMPRESSION)
+        listenerWeakRef.get()?.onPartnerAdImpression(partnerAdWeakRef.get() ?: return)
+            ?: PartnerLogController.log(
+                CUSTOM,
+                "Unable to fire onPartnerAdImpression for Google Bidding adapter.",
+            )
+    }
+
+    override fun onAdFailedToShowFullScreenContent(adError: AdError) {
+        PartnerLogController.log(SHOW_FAILED, adError.message)
+        continuationWeakRef.get()?.let {
+            if (it.isActive) {
+                it.resume(
+                    Result.failure(
+                        ChartboostMediationAdException(
+                            GoogleBiddingAdapter.getChartboostMediationError(adError.code),
+                        ),
+                    ),
+                )
+            }
+        }
+    }
+
+    override fun onAdShowedFullScreenContent() {
+        PartnerLogController.log(SHOW_SUCCEEDED)
+
+        continuationWeakRef.get()?.let {
+            if (it.isActive) {
+                it.resume(Result.success(partnerAdWeakRef.get() ?: return))
+            }
+        }
+    }
+
+    override fun onAdClicked() {
+        PartnerLogController.log(DID_CLICK)
+
+        listenerWeakRef.get()?.onPartnerAdClicked(partnerAdWeakRef.get() ?: return)
```

Comment:
also not return. we should still log. similar to a bunch of these so I'll just mark it here once.